frontend/public/components/app-contents.tsx

```diff
@@ -80,21 +80,57 @@ const LazyRoute = (props) => (
   />
 );
 
-const getPluginPageRoutes = (activePerspective: string, routePages: RoutePage[]) =>
-  routePages.map((r) => {
-    if (r.properties.perspective && r.properties.perspective !== activePerspective) {
-      return null;
-    }
-    const Component = r.properties.loader ? LazyRoute : Route;
-    return <Component {...r.properties} key={Array.from(r.properties.path).join(',')} />;
-  });
+const getPluginPageRoutes = (
+  activePerspective: string,
+  setActivePerspective: (perspective: string) => void,
+  routePages: RoutePage[],
+) => {
+  const activeRoutes = routePages
+    .filter((r) => !r.properties.perspective || r.properties.perspective === activePerspective)
+    .map((r) => {
+      const Component = r.properties.loader ? LazyRoute : Route;
+      return <Component {...r.properties} key={Array.from(r.properties.path).join(',')} />;
+    });
+
+  const inactiveRoutes = routePages
+    .filter((r) => r.properties.perspective && r.properties.perspective !== activePerspective)
+    .map((r) => {
+      const key = Array.from(r.properties.path).join(',');
+      return r.properties.loader ? (
+        <LazyRoute
+          {...r.properties}
+          key={key}
+          loader={() => {
+            setActivePerspective(r.properties.perspective);
+            return r.properties.loader();
+          }}
+        />
+      ) : (
+        <Route
+          {...r.properties}
+          key={key}
+          component={undefined}
+          render={(componentProps) => {
+            setActivePerspective(r.properties.perspective);
+            return r.properties.component ? (
+              <r.properties.component {...componentProps} />
+            ) : (
+              r.properties.render(componentProps)
+            );
+          }}
+        />
+      );
+    });
+
+  return [...activeRoutes, ...inactiveRoutes];
```

Comment:
You must render all `inactiveRoutes` after all dynamic and static routes but before the fallback route.

This will ensure that we aren't pushing the user into the wrong perspective even though there's a static route available.

Reply:
Updated.